indico/modules/events/surveys/controllers/management.py

```diff
@@ -264,3 +267,58 @@ def _render_questionnaire(survey):
     tpl = get_template_module('events/surveys/_questionnaire.html')
     form = make_survey_form(survey.questions)()
     return tpl.render_questionnaire(survey, form)
+
+
+class RHExportSubmissions(RHManageSurveyBase):
+    """Export submissions from the survey to a CSV file"""
+
+    CSRF_ENABLED = False
+
+    def _process(self):
+        if not self.survey.submissions:
+            flash(_('There are no submissions in this survey'))
+            return redirect(url_for('.manage_survey', self.survey))
+
+        submission_ids = set(map(int, request.form.getlist('submission_ids')))
+        csv_file = generate_csv_from_survey(self.survey, submission_ids)
+        delete_file.apply_async(args=[csv_file.name], countdown=3600)
+        return send_file('submissions-{}.csv'.format(self.survey.id), csv_file.name, 'text/csv')
+
+
+class RHSurveySubmissionBase(RHManageSurveysBase):
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.submission
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageSurveysBase._checkParams(self, params)
+        survey_strategy = joinedload('survey')
+        answers_strategy = defaultload('answers').joinedload('question')
+        self.submission = (SurveySubmission
+                           .find(id=request.view_args['submission_id'])
+                           .options(answers_strategy, survey_strategy)
+                           .one())
+
+
+class RHDeleteSubmissions(RHManageSurveyBase):
+    """Remove submissions from the survey"""
+
+    def _process(self):
+        submission_ids = map(int, request.form.getlist('submission_ids'))
+        for submission_id in submission_ids:
```

Comment:
Right now you could delete submissions from any survey, even in another event you don't have access to :volcano: